### ISSUE TITLE
Unable to disable application level security on APIM 3.0.0

### ISSUE THREAD

**Issue by 321pranay** [wso2/product-apim#7511]
### Description:
In the APIM Publisher > API >Runtime Configuration, Application Level Security has Mandatory and Optional fields which are blocked.
![image](https://user-images.githubusercontent.com/20185958/74905339-f44b4580-53d3-11ea-8e74-573793e5fd85.png)

Also in APIM Developer Portal , there is no option to create an application without token type.
![image](https://user-images.githubusercontent.com/20185958/74905525-7176ba80-53d4-11ea-9bd8-2c4fa04588e1.png)
![image](https://user-images.githubusercontent.com/20185958/74905578-86ebe480-53d4-11ea-9dcc-ced04fcde855.png)

### Steps to reproduce:
1. Open an API in APIM Publisher portal.
2. Create an application in APIM Developer portal

### Affected Product Version:
3.0.0

### Environment details (with versions):
- OS: Windows 10
- Client: APIM
- Env (Docker/K8s): Installer

**Comment by tmkasun** [comment]
Hi @321pranay 

You can't disable security for the entire API. `Optional` and `Mandatory` options in the application level security section are for giving the preference between Application and Trasport level security types.

For example, If you enable both security schemes in Transport level and application-level security, You can specify one out of them as optional, but not both Transport level and application-level 

If you want to disable security for entire API, You have to switch off security for individual operations(resource) from the APIM Publisher > API > Resources page

After disabling the security, You can invoke API without giving any security credentials (I:e no need to give access token in `Authorization: Bearer` header)

@erangatl , I think we have an UX issue here, users get confused about mandatory and optional fields and how to completely disable security, WDYT ?

**Comment by erangatl** [comment]
Yes this field needs to be shown only when Mutual SSL and Application level security is enabled. 
Else uses get confused.

**Comment by jadam1212** [comment]
Hi, this way of disabling security doesn't seem to work for SOAP APIs. I am trying to do the same but it just keeps complaining about no credentials provided (403). It seems like a bug for 3.0.0, can you please confirm?

**Comment by tmkasun** [comment]
Hi @jadam1212 ,

Are you getting `403` error only when changing the security scheme?
How did you know it's a 403 ? Did you check the browser developer console network activities?
Can you share a screenshot of the error you are getting?

**Comment by mancave** [comment]
If security is needed on SOAP, Mutual SSL & BA should be there a transport level. Application authorization should be done with Web Services Security UsernameToken Profile 1.0 Specification. The Web Services Security UsernameToken Profile 1.0 Specification can be transformed to a BA impl. This needed to be implemented on 3.x.x
